src/main/java/uk/ac/ebi/eva/pipeline/jobs/steps/processors/AnnotationParserProcessor.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2015-2017 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.eva.pipeline.jobs.steps.processors;
+
+import org.springframework.batch.item.ItemProcessor;
+
+import uk.ac.ebi.eva.commons.models.mongo.entity.Annotation;
+import uk.ac.ebi.eva.pipeline.io.VepProcess;
+import uk.ac.ebi.eva.pipeline.io.mappers.AnnotationLineMapper;
+import uk.ac.ebi.eva.pipeline.parameters.AnnotationParameters;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * ItemStreamWriter that takes VariantWrappers and serialize them into a {@link VepProcess}, which will be responsible
+ * for annotating the variants and writing them to a file.
+ */
+public class AnnotationParserProcessor implements ItemProcessor<List<String>, List<Annotation>> {
+
+    private static final int UNUSED_LINE_NUMBER = 0;
+
+    private AnnotationLineMapper annotationLineMapper;
+
+    public AnnotationParserProcessor(AnnotationParameters annotationParameters) {
+        annotationLineMapper = new AnnotationLineMapper(annotationParameters.getVepVersion(),
+                                                        annotationParameters.getVepCacheVersion());
+    }
+
+    @Override
+    public List<Annotation> process(List<String> ensemblVariants) throws Exception {
+        List<Annotation> annotations = new ArrayList<>();
+        for (String ensemblVariant : ensemblVariants) {
```

Comment:
This is not bad, but it seems that this loop could be replaced by something like:
ensemblVariants.stream().map(v -> annotationLineMapper.mapLine(v, ....)).collect(Collectors.toList());